src/proxy/inbound.rs

```diff
@@ -132,9 +132,12 @@ impl Inbound {
                         }
                         Hbone(req) => match hyper::upgrade::on(req).await {
                             Ok(mut upgraded) => {
-                                super::copy_hbone("hbone server", &mut upgraded, &mut stream)
-                                    .await
-                                    .expect("hbone server copy");
+                                let res =
+                                    super::copy_hbone("hbone server", &mut upgraded, &mut stream)
+                                        .await;
+                                if res.is_err() {
+                                    error!("hbone server copy: {:?}", res);
+                                }
```

Comment:
```suggestion
                               if let Err(e) = super::copy_hbone("hbone server", &mut upgraded, &mut stream)
                                    .await {
                                    error!("hbone server copy: {}", e);
                                }
```

Is probably preferred. Then we don't need to user debug formatter.

Reply:
Yes, this is very clean.